### PR TITLE
[wip] Neighborfiles util imple

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## TODO
1. function documentation
2. dedupe duplicate code
3. PR summary
4. remove "workspaceFolder" from function argument of `neighborFiles` & `listFilesWithinDistanceAgainstFile`, passing them for only making tests working at this point of time
5. fix https://github.com/aws/aws-toolkit-vscode/pull/5319/files#diff-983eadb11c85ec3bbe768f8dece2e5962e35b06fadab6db553c4e915d8517687R702

## Problem

jb pr https://github.com/aws/aws-toolkit-jetbrains/pull/4673

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/core/src/shared/utilities/workspaceUtils.ts [hunk 1]
```diff
@@ -604,3 +604,112 @@ export async function collectFilesForIndex(
     // pick top 100k files below size limit
     return storage.slice(0, Math.min(100000, i))
 }
+
+export async function listFilesWithinDistanceAgainstFile(
```
Comment (justinmk3) [inline packages/core/src/shared/utilities/workspaceUtils.ts:608]: side-note: IIRC, codewhisper (used to?) have similar functionality for its "dependency graph" stuff. I know that "dependency graph" was removed; was its "nearby files" logic also removed?

Comment (Will-ShaoHua) [reply packages/core/src/shared/utilities/workspaceUtils.ts:608]: umm, don't think we have them anymore but i will double check this Justin

Comment (Will-ShaoHua) [reply packages/core/src/shared/utilities/workspaceUtils.ts:608]: > have similar functionality for its "dependency graph" stuff


and i feel previously code scan use case is to search files with dependency but not file distance

Comment (Will-ShaoHua) [reply packages/core/src/shared/utilities/workspaceUtils.ts:608]: but i will double check this, Justin

File: packages/core/src/awsService/accessanalyzer/vue/iamPolicyChecks.vue [hunk 2]
```diff
@@ -447,15 +447,15 @@ export default defineComponent({
...
+                .then((response) => {
                     this.initialData.checkAccessNotGrantedActionsTextArea = response.actions
                         ? response.actions.toString()
                         : ''
                     this.initialData.checkAccessNotGrantedResourcesTextArea = response.resources
                         ? response.resources.toString()
                         : ''
                 })
-                .catch(err => console.log(err))
+                .catch((err) => console.log(err))
```
Comment (Will-ShaoHua) [inline packages/core/src/awsService/accessanalyzer/vue/iamPolicyChecks.vue:458]: looks like changes here are made by linter

**Comment by Will-ShaoHua** [comment]
closing and replacing the PR with #5582 targeting feature branch q-inline-crossfile

**Pull request closed without merging**